# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1512585567 21600
#      Wed Dec 06 12:39:27 2017 -0600
# Branch stable
# Node ID ccedd8a29d3e14ae704732176863e1d616bb401c
# Parent  9cba59ee95644a17970e280ec2ad4d7260664f1e
Make gnuplot color have three components for interpolated edge color (bug #52595).

* __gnuplot_draw_axes__.m: Map the color data into a three-component eight bit
  integer for the interpolated edgecolor scenario.

diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -949,17 +949,17 @@ function __gnuplot_draw_axes__ (h, plot_
                       ccol = repmat (ccol, numel (xcol), 1);
                     endif
                     color = "flat";
                     have_cdata(data_idx) = true;
                   endif
                 elseif (strcmp (ec, "interp"))
                   if (numel (ccol) == 3)
                     warning ("\"interp\" not supported, using 1st entry of cdata");
-                    color = ccol(1,:);
+                    color = mapcdata (ccol(:), obj.cdatamapping, clim, cmap_sz);
                   else
                     if (isscalar (ccol))
                       ccol = repmat (ccol, numel (xcol), 1);
                     endif
                     color = "interp";
                     have_cdata(data_idx) = true;
                   endif
                   ccol = mapcdata (ccol, obj.cdatamapping, clim, cmap_sz);
